ci/build.sh

```diff
@@ -4,12 +4,83 @@ set -e -u -x -o pipefail
 
 cd "$(dirname "$0")/.."
 
-PLATFORM=""
+if [[ -z ${BUILDKITE+x} ]]; then
+  echo "This script is intended to run in CI only."
+  exit 1
+fi
+
+SPATIALOS_TOOLBELT_VERSION="latest"
+
+TOOLBELT_PLATFORM=""
+SPATIAL_PLATFORM=""
+SPATIAL_BINARY=""
+
+# Setup platform specific environment variables.
 if [[ "$(uname -s)" == "Linux" ]]; then
-  PLATFORM="linux"
+  TOOLBELT_PLATFORM="linux"
+  SPATIAL_PLATFORM="linux"
+  SPATIAL_BINARY="spatial"
 elif [[ "$(uname -s)" == "Darwin" ]]; then
-  PLATFORM="macos"
+  TOOLBELT_PLATFORM="mac"
+  SPATIAL_PLATFORM="macos"
+  SPATIAL_BINARY="spatial"
+
+  imp-tool-bootstrap subscribe \
+    --use_gcs_oidc_auth=false \
+    --gcs_credentials_type=auto \
+    --tools=imp-tool
+else
+  TOOLBELT_PLATFORM="win"
+  SPATIAL_PLATFORM="windows"
+  SPATIAL_BINARY="spatial.exe"
+fi
+
+# Create temporary directories for the spatial binary and authentication token.
+export TOOLS_DIR="$(pwd)/tools"
```

Comment:
Use `mktemp -d` instead of this + the trap below.

Reply:
Done